### DISCUSSION TITLE
ServiceClient External Token Management with User Azure Managed Identity taking 8 seconds to get Access Token

### DISCUSSION THREAD

**Discussion by twiga2013** [microsoft/PowerPlatform-DataverseServiceClient#357] [Q&A]
Hi,

I have an implementation of using ServiceClient with external token management using the DefaultAzureCredential from Azure.Identity taking around 8 seconds to get authenticated from Dataverse. I am not sure why its taking that long and if there ways to bring the time down. 

 AccessToken newAccessToken = await _defaultCredential.GetTokenAsync(new TokenRequestContext(new[] { $"{environment}/.default" }));

were environment is the Dataverse URL.

**Comment by MattB-msft** [comment]
if you enabled verbose logging, you will be able to see what the client is taking time on.
Given your description, It sounds like your connecting to a 'cold' environment.  when the client first connects to and environment it queries that environment for its configuration information to allow the client to scale its operation based on the version of Dataverse its talking to. In a 'cold' environment or a 'cold' node, this can take a few seconds as the environment is woken up and loaded into server cache. 

The version we just release reorganizes the login flow and makes some assumptions about version to speed up login.  The effect is a 'faster' client create and the first call your code makes shows the perf hit if the server cache has to be loaded.
